magit-gh-comments-github.el

```diff
@@ -110,42 +128,127 @@ element of the result is an alist with the following keys:
-                              (reverse
-                               (dolist (cell alist result)
-                                 (setq result
-                                       (cons
-                                        (cons (funcall fn (car cell))
-                                              (cdr cell))
-                                        result)))))))
-    (let ((comment-fields '(body position author path))
-          (comments (magit-gh--request-sync
-                     (magit-gh--url-for-pr-comments pr)
-                     :headers '(("Authorization" . (format "token %s" (magit-gh--get-oauth-token))))
-                     :parser (lambda ()
-                               (let ((json-array-type 'list))
-                                 (json-read-array)))))
-          result)
-      (dolist (comment (remove-if #'outdated-p comments))
-        (let* ((author (alist-get 'login (alist-get 'user comment)))
-               (comment (add-to-list 'comment `(author . ,author))))
-          (setq result (cons (alist-filter comment-fields comment)
-                             result))))
-      (reverse (mapcar
-                (lambda (x) (alist-map-keys #'sym-to-keyword x))
-                result)))))
+  (apply 'append
+         (mapcar (lambda (review) (alist-get :comments review))
+                 (magit-gh--list-reviews pr))))
+
 
 ;; TODO: Make an integration test out of this:
 ;; (magit-gh--list-comments magit-gh-comment-test-pr)
+;; (magit-gh--list-reviews magit-gh-comment-test-pr)
+
+(defun magit-gh--assoc-recursive (keys alist)
+  (cond ((not (listp keys)) (magit-gh--assoc-recursive (list keys) alist))
+        ((not keys) nil)
+        ((not (listp alist)) nil)
+        ((not (cdr keys)) (assoc (car keys) alist))
+        (t (magit-gh--assoc-recursive (cdr keys) (cdr (assoc (car keys) alist))))))
+
+
+(defun magit-gh--parse-json-array ()
+ (let ((json-array-type 'list))
+   (json-read-array)))
+
+(defun magit-gh--keys->keywords (alist)
+  "Transform the keys in ALIST into colon-prefixed symbols."
+    (cl-flet ((sym-to-keyword (x)
+                              (intern (format ":%s" x)))
+              (alist-map-keys (fn alist)
+                              (let (result)
+                                (reverse
+                                 (dolist (cell alist result)
+                                   (setq result
+                                         (cons
+                                          (cons (funcall fn (car cell))
+                                                (cdr cell))
+                                          result)))))))
+        (mapcar (lambda (x) (alist-map-keys #'sym-to-keyword x))
+                alist)))
+
+(defun magit-gh--alist-filter (keys alist)
+  (-filter 'identity ;; remove nils
+          (mapcar (lambda (k) (magit-gh--assoc-recursive k alist))
+                  keys)))
+
+(defun magit-gh--response-parser-backup (fields-of-interest)
```

Comment:
Don't remember what the plan for this was, but now that it is in source-control let's remove it.